include/aspect/simulator.h

```diff
@@ -594,6 +594,8 @@ namespace aspect
       void
       local_assemble_advection_system (const AdvectionField &advection_field,
                                        const std::pair<double,double> global_field_range,
+                                       const bool use_smoothing, \
+                                       Vector<float> viscosity_per_cell, \
```

Comment:
Fixed, and I would still need the boolean "use_smoothing" inside this function because when it gets to "nu" I don't want it to do what it normally does, I want it to use what is inside "viscosity_per_cell".  I can make the averaging (or taking max) happen outside of this though.